application/ui/components/input/custom-object.jsx

```diff
@@ -0,0 +1,206 @@
+/** @jsx React.DOM */
+'use strict';
+
+var _             = require('underscore');
+var React         = require('react');
+var ObjectParameter;
+
+module.exports = ObjectParameter = React.createClass({
+
+    displayName : 'ObjectParameter',
+
+    getInitialState : function()
+    {
+        return {
+            values : []
+        };
+    },
+
+    getValue : function()
+    {
+        return this.state.values;
+    },
+
+    getDefaultValue : function()
+    {
+        return {
+            key: '',
+            value: '',
+            inputType: 'string'
+        };
+    },
+
+    getInputs : function()
+    {
+        var inputs   = [],
+            instance = this;
+
+        _.each(this.state.values, function (value, index) {
+            inputs.push(instance.getInput(value, index));
+        });
+
+        return inputs;
+    },
+
+    getInput : function(prop, index)
+    {
+        var field, selectType, instance = this;
+
+        if (prop.inputType !== 'object') {
+            field = (
+                <input
+                    className    = 'array-input'
+                    value        = {prop.value}
+                    onChange     = {_.partial(instance.updateField, index, 'value')}
+                />
+            );
+        } else {
+            field = (
+                <table>
+                    <ObjectParameter value={prop.value} index={index} onChange={_.partial(instance.updateField, index, 'value')}/>
+                </table>
+            );
+        }
+
+        var rmCallback = function () {
+            instance.removeField(index);
+        };
+
+        if (prop.inputType === 'object') {
+            selectType = 'object';
+        } else {
+            selectType = (
+                <select
```

Comment:
There's a `SelectInput` component in `ui/components/input` that would allow for less markup here. (Not having to manually put the `option` tags.)